doc/evaluation_design.md

```diff
@@ -0,0 +1,70 @@
+    <img src="figures/train_and_evaluate_flowchart_proposal.svg"
+    <br>
+    <em>Figure 3 </em>. Proposal Flowchart of Train and Evaluation </div>
+</center>
+
+2. Pserver stops to update model variables when a evaluation job starts.\
+As shown in <em>Figure 1</em>, pserver will not execute the 17th and 18th steps after a evaluation job starts at the 10th step. So, the model worker 2 get is same as the worker 1. The solution can resolve Problem 2 and also can resolve Problem 1 by solution in <em>Figure 3.</em>. But, the train process must wait all evaluationt tasks have been completed. 
+
+Questions to discuss:
+* None of training checkpoints will be evaluted in above two solutions which is different from tf.estimator. tf.estimator will load the lastest checkpoint saved by train to evaluate in a separated pod named evaluator. 
+* Should master preserve the model to checkpoint after evaluation?
+
+## Evaluate by Loading Checkpoint to Pserver
+Generally, deep learning framework preserves model as checkpoint every many seconds or steps. So pserver can load model from checkpoint for evaluation. There are also two designs for pserver to load model. 
+
+1. The master launches additional pods as evalution pserver.\
+When a evaluation job starts, the master will launch additional pods as evaluation pservers and evaluation pservers will load model from checkpoint.\
+The worker gets model from evaluation pservers to inference outputs for evluation tasks and reports outputs to master.\
+The master will release those pods after receiving all outputs of evalution tasks from worker.\
+In this case, evaluation is separated from train. So, some workers can process evaluation tasks and other workers can process train tasks at same time. But more additional pods are needed.
+
+2. Existing pservers stop train task and load model from checkpoint to evaluate.\
+When a evaluation job starts, pservers preserve the current model to temporary file and load model from the lastest checkpoint. After evaluation, pservers will restore model from temporary file preserved and continue to train. \
+The design of pserver will be complicated in this case. What's more,some workers may be reporting gradients when the master change model to evaluation for psersers and pservers can not update model by those gradients at this moment.
+
+
+## Introduction of tf.estimator to Evaluate
+tf.estimator will launch pods as pservers, workers and evalutor when a distributed train job is submitted with ParameterServerStrategy. The evaluator does not participate in train. tf.estimator decides when to evaluate by [throttle_secs](https://www.tensorflow.org/api_docs/python/tf/estimator/EvalSpec#throttle_secs). tf.estimator will check whether there are new checkpoints in checkpoint directory every throttle_secs after the last evaluation ends. The evaluator will restore whole model from the lastest checkpoint to evaluate. So, evaluaion of tf.estimator is not distributed and not fault-tolerant.
```

Comment:
The evaluator only have one instance. It will restore the whole model from the lastest checkpoint.